src/main/java/core/AbstractGameState.java

```diff
@@ -202,7 +221,7 @@ public final AbstractGameState copy() {
      * @return - double, score of current state.
      */
     public final double getScore(int playerId) {
-        return _getScore(playerId);
+        return _getHeuristicScore(playerId);
```

Comment:
A problem here. It's good to split the heuristic and game score, yes. But a function named "getScore" becomes ambiguous then, and given this split I'd actually expect it to return the game score.

Reply:
I agree. In this case I was not aggressive enough, and didn;t want to change the name of the existing method. Now I have permission I'll refactor this!